scripts/Diffraction/isis_powder/routines/absorb_corrections.py

```diff
@@ -112,3 +112,29 @@ def _do_cylinder_absorb_corrections(ws_to_correct, multiple_scattering, is_vanad
     if previous_units != ws_units.tof:
         ws_to_correct = mantid.ConvertUnits(InputWorkspace=ws_to_correct, Target=previous_units)
     return ws_to_correct
+
+
+def apply_paalmanpings_absorb_and_subtract_empty(workspace, summed_empty, sample_details,
+                                                 paalman_pings_events_per_point=None):
+    container_geometry = sample_details.generate_container_geometry()
+    container_material = sample_details.generate_container_material()
+    if container_geometry and container_material:
+        mantid.SetSample(workspace, Geometry=sample_details.generate_sample_geometry(),
+                         Material=sample_details.generate_sample_material(),
+                         ContainerGeometry=container_geometry,
+                         ContainerMaterial=container_material)
+    else:
+        mantid.SetSample(workspace, Geometry=sample_details.generate_sample_geometry(),
+                         Material=sample_details.generate_sample_material())
```

Comment:
I'm not sure this code path works - if a container isn't specified (eg if I comment out the sample_details.set_container line in the test script) I get an error from the ApplyPaalmanPingsCorrection line:
`Error in execution of algorithm ApplyPaalmanPingsCorrection:
'acc'
  at line 89 in 'C:/mantid/Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ApplyPaalmanPingsCorrection.py'
`
I think if the container isn't set then the CanWorkspace parameter shouldn't be supplied lower down on L138

This only happens if someone specifies empty_can_subtraction_method=PaalmanPings but then doesn't set up a container. But worth fixing up I think